src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java

```diff
@@ -202,4 +204,110 @@ void deleteNoCompanyProfile() {
         verify(repository, never()).delete(any());
     }
 
+    @Test
+    void createDissolvedCompany() {
+        final Address mockServiceAddress = new Address(
+                "","","","","",""
+        );
+        CompanySpec spec = new CompanySpec();
+        spec.setCompanyNumber(COMPANY_NUMBER);
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus(COMPANY_STATUS_DISSOLVED);
+
+        CompanyProfile savedProfile = new CompanyProfile();
+        when(randomService.getEtag()).thenReturn(ETAG);
+        when(repository.save(any())).thenReturn(savedProfile);
+        when(addressService.getAddress(spec.getJurisdiction())).thenReturn(mockServiceAddress);
+        CompanyProfile returnedProfile = this.companyProfileService.create(spec);
+
+        assertEquals(savedProfile, returnedProfile);
+
+        ArgumentCaptor<CompanyProfile> companyProfileCaptor = ArgumentCaptor.forClass(CompanyProfile.class);
+        verify(repository).save(companyProfileCaptor.capture());
+
+        CompanyProfile profile = companyProfileCaptor.getValue();
+        assertEquals(COMPANY_NUMBER, profile.getId());
+        assertEquals(COMPANY_NUMBER, profile.getCompanyNumber());
```

Comment:
Hi @lyadavalli you didn't respond to this one?

Reply:
updated based on comments

Reply:
there still seems to be a lot of duplicated code here - when we add more tests this will keep growing.  Can you not just move all the common assertions into a single method and call it from each test?

Reply:
created common method for company profile assertions